packages/x-date-pickers/src/internals/hooks/useField/useField.utils.ts

```diff
@@ -253,75 +254,101 @@ export const addPositionPropertiesToSections = <TSection extends FieldSection>(
 
-  return utils.formatByString(date, format);
+    default: {
+      return currentTokenValue;
+    }
+  }
 };
 
 export const splitFormatIntoSections = <TDate>(
   utils: MuiPickerFieldAdapter<TDate>,
+  localeText: PickersLocaleText<TDate>,
   format: string,
   date: TDate | null,
 ) => {
   let currentTokenValue = '';
   const sections: Omit<FieldSection, 'start' | 'end'>[] = [];
   const expandedFormat = utils.expandFormat(format);
 
+  const commitCurrentToken = () => {
+    if (currentTokenValue === '') {
+      return;
+    }
+
+    const sectionConfig = getDateSectionConfigFromFormatToken(utils, currentTokenValue);
+    const sectionValue = date == null ? '' : utils.formatByString(date, currentTokenValue);
+
+    sections.push({
+      ...sectionConfig,
+      formatValue: currentTokenValue,
+      value: sectionValue,
+      placeholder: getSectionPlaceholder(utils, localeText, sectionConfig, currentTokenValue),
+      separator: '',
+      edited: false,
+    });
+
+    currentTokenValue = '';
+  };
+
   for (let i = 0; i < expandedFormat.length; i += 1) {
     const char = expandedFormat[i];
-    if (!char.match(/([A-zÀ-ú]+)/g)) {
-      if (currentTokenValue === '') {
-        sections[sections.length - 1].separator += char;
-      } else {
-        const dateForCurrentToken = formatDateWithPlaceholder(utils, date, currentTokenValue);
-        if (dateForCurrentToken === currentTokenValue) {
-          sections[sections.length - 1].separator += currentTokenValue;
-          currentTokenValue = '';
-        } else {
-          const sectionConfig = getDateSectionConfigFromFormatToken(utils, currentTokenValue);
-
-          sections.push({
-            ...sectionConfig,
-            formatValue: currentTokenValue,
-            value: dateForCurrentToken,
-            emptyValue: sectionConfig.dateSectionName,
-            separator: char,
-            edited: false,
-          });
-          currentTokenValue = '';
-        }
-      }
-    } else {
-      currentTokenValue += char;
-    }
 
-    if (i === expandedFormat.length - 1) {
-      const dateForCurrentToken = formatDateWithPlaceholder(utils, date, currentTokenValue);
-      if (dateForCurrentToken === currentTokenValue) {
-        sections[sections.length - 1].separator += currentTokenValue;
-      } else {
-        const sectionConfig = getDateSectionConfigFromFormatToken(utils, currentTokenValue);
-
-        sections.push({
-          ...sectionConfig,
-          formatValue: currentTokenValue,
-          value: dateForCurrentToken,
-          emptyValue: sectionConfig.dateSectionName,
-          separator: null,
-          edited: false,
-        });
+    if (char.match(/([A-zÀ-ú]+)/g)) {
+      currentTokenValue += char;
+    } else {
+      if (currentTokenValue !== '') {
+        commitCurrentToken();
       }
+
+      sections[sections.length - 1].separator += char;
     }
   }
 
-  return sections;
+  commitCurrentToken();
+
+  return sections.map((section) => ({
```

Comment:
Replaces the `/` separator by ` / `

Advantages:
- The default placeholder of `DateField` (`L` for `dayjs`) is more readable `MM/DD/YYYY` becomes `DD / MM / YYYY`
- Matches the design proposed by @gerdadesign on https://github.com/mui/mui-x/issues/5531#issuecomment-1226091731

Disadvantages:
- Users can't remove this behavior so if they want something more compact they can't